csharp/src/Ice/Communicator.cs

```diff
@@ -1404,6 +1446,216 @@ private static void LoadReferencedAssemblies(Assembly a)
+
+                    entryPoint = args[0];
+
+                    args = args.Skip(1).ToArray();
+
+                    // Convert command-line options into properties. First we convert the options from the plug-in
+                    // configuration, then we convert the options from the application command-line.
+                    var properties = new Dictionary<string, string>();
+                    properties.ParseArgs(ref args, name);
+                    properties.ParseArgs(ref cmdArgs, name);
+                    foreach (KeyValuePair<string, string> p in properties)
+                    {
+                        SetProperty(p.Key, p.Value);
+                    }
+                }
+                Debug.Assert(entryPoint != null);
+                // Always check the static plugin factory table first, it takes precedence over the entryPoint specified
+                // in the plugin property value.
+                if (!_pluginFactories.TryGetValue(name, out IPluginFactory? pluginFactory))
+                {
+                    // Extract the assembly name and the class name.
+                    int sepPos = entryPoint.IndexOf(':');
+                    if (sepPos != -1)
+                    {
+                        const string driveLetters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+                        if (entryPoint.Length > 3 &&
+                           sepPos == 1 &&
+                           driveLetters.IndexOf(entryPoint[0]) != -1 &&
+                           (entryPoint[2] == '\\' || entryPoint[2] == '/'))
+                        {
+                            sepPos = entryPoint.IndexOf(':', 3);
+                        }
+                    }
+                    if (sepPos == -1)
+                    {
+                        throw new FormatException($"error loading plug-in `{entryPoint}': invalid entry point format");
+                    }
+
+                    System.Reflection.Assembly? pluginAssembly;
+                    string assemblyName = entryPoint[..sepPos];
+                    string className = entryPoint[(sepPos + 1)..];
+
+                    try
+                    {
+                        // First try to load the assembly using Assembly.Load, which will succeed if a fully-qualified
+                        // name is provided or if a partial name has been qualified in configuration. If that fails, try
+                        // Assembly.LoadFrom(), which will succeed if a file name is configured or a partial name is
+                        // configured and DEVPATH is used.
+                        //
+                        // We catch System.Exception as this can fail with System.ArgumentNullException or
+                        // System.IO.IOException depending of the .NET framework and platform.
+                        try
+                        {
+                            pluginAssembly = System.Reflection.Assembly.Load(assemblyName);
```

Comment:
`System.Reflection.` is no needed in this file, we have a using directive for it.

Reply:
Fixed.